### ISSUE TITLE
rke2-calico cni install does not honor global.systemDefaultRegistry

### ISSUE THREAD

**Issue by ron1** [rancher/rke2#1236]
**Environmental Info:**
RKE2 Version: v1.21.2+rke2r1
Install via rke2.linux-amd64.tar.gz

Node(s) CPU architecture, OS, and Version: 3.10.0-1160.21.1.el7.x86_64

Cluster Configuration: 3 servers, 6 agents

**Describe the bug:**
Calico CNI installation does not pull Calico images from custom system-default-registry. On servers, mirrored-calico-(kube-controllers|pod2daemon-flexvol|typha) images were incorrectly pulled from docker.io. On agents, mirrored-calico-(cni|pod2daemon-flexvol|node) were also pulled from docker.io. All other images on the nodes were correctly pulled from the custom system-default-registry.

**Steps To Reproduce:**

- Installed RKE2: via github.com/rancherfederal/rke2-ansible tarball_install with inventory/sample/group_vars/rke2_(agents|servers).yml updated with:
```
rke2_config:
  cni: calico
  system-default-registry: "<my_system_default_registry>"
...
```
**Expected behavior:**
Running `ctr container ls | grep docker | wc -l` on servers/agents returns 0

**Actual behavior:**
Running `ctr container ls | grep docker | wc -l` on servers/agents returns 3

**Additional context / logs:**

**Comment by brandond** [comment]
@manuelbuil it looks like we use Values.global.systemDefaultRegistry for the operator image; is it also plumbed through into the other images?

**Comment by manuelbuil** [comment]
> @manuelbuil it looks like we use Values.global.systemDefaultRegistry for the operator image; is it also plumbed through into the other images?

No, it is not. I should have done this once the imagePath and imagePrefix was available but I forgot. This is a real bug that must be fixed. I'll prepare a fix for this

**Comment by manuelbuil** [comment]
Not sure how this got closed... :thinking:

**Comment by brandond** [comment]
You said the magic words "to fix #xx" in the PR message :)

**Comment by rancher-max** [comment]
This has been validated in the 1.21 line previously, but we should ensure to test airgap with calico when doing 1.22 validations once we have an rc available.
